Orbitersdk/samples/ProjectApollo/src_sys/PanelSDK/Internals/esysparse.cpp

```diff
@@ -163,6 +163,80 @@ void E_system::Create_Diode(char *line)
 	AddSystem(new Diode(name, Source, NominalTemperature, SaturationCurrent));
 }
 
+void E_system::Create_ElectricLight(char* line)
+{
+	/* Config file creation string is as follows
+	
+	A single line begining with "<LIGHT>"
+
+	Followed by the fields:
+
+			Field			|			Unit			|			Type			|
+	---------------------------------------------------------------------------------
+		name						n/a							string
+		sourceName					n/a							string
+		flashing					n/a							string
+		OnTime						seconds						double
+		OffTime						seconds						double
+		<position>					meters						VECTOR3
+		<direction>					meters						VECTOR3
+		CutoffRange					meters						double
+		ConstantAttenuation			meters						double
+		LinearAttenuation			meters^-1					double
+		QuadraticAttenuation		meters^-2					double
+		Umbra						radians						double
+		Penumbra					radians						double
+		<%f %f %f %f>				n/a							COLOUR4
+		<%f %f %f %f>				n/a							COLOUR4
+		<%f %f %f %f>				n/a							COLOUR4
+		%lf							Watts						double
+		%lf							Volts						double
+	
+	
+	Example:
+
+	<LIGHT> SPOTLIGHT DC_DUMMY 0 0.0 0.0 <-1.439 1.390 0.920> <0.0045 -0.0046 0.9999> 5000.0 0.0 0.0 0.0015 0.174533 0.5 <1.0 0.945 0.878 0.0> <1.0 0.945 0.878 0.0> <0.0 0.0 0.0 0.0> 100.0 115.0
+
+
+	*/ 
+
+
+	char name[100];
+	char sourceName[100];
+	e_object* powerSource = nullptr;
+	bool flashing = false;
+	double onTime = 0;
+	double offTime = 0;
+	VECTOR3 pos = _V(0, 0, 0);
+	VECTOR3 dir = _V(1, 0, 0);
+	double range = 0;
+	double att0 = 0;
+	double att1 = 0;
+	double att2 = 0;
+	double umbra = 0;
+	double penumbra = 0;
+	COLOUR4 diffuse = { 0, 0, 0, 0};
+	COLOUR4 specular = { 0, 0, 0, 0 };
+	COLOUR4 ambient = { 0, 0, 0, 0 };
+	double powerDraw = 0;
+	double nomVoltage = 0;
+
+	sscanf(line + 7, "%s %s %d %lf %lf <%lf %lf %lf> <%lf %lf %lf> %lf %lf %lf %lf %lf %lf <%f %f %f %f> <%f %f %f %f> <%f %f %f %f> %lf %lf",
+		&name, &sourceName, &flashing, &onTime, &offTime,
+		&pos.x, &pos.y, &pos.z,
+		&dir.x, &dir.y, &dir.z,
+		&range, &att0, &att1, &att2, &umbra, &penumbra, 
+		&diffuse.r, &diffuse.g, &diffuse.b, &diffuse.a,
+		&specular.r, &specular.g, &specular.b, &specular.a, 
+		&ambient.r, &ambient.g, &ambient.b, &ambient.a,
+		&powerDraw, &nomVoltage);
+
+
+	powerSource = (e_object*)GetPointerByString(sourceName);
+	ElectricLight* newLight = new ElectricLight(name, powerSource, flashing, onTime, offTime, (VESSEL*)(this->Vessel), pos, dir, range, att0, att1, att2, umbra, penumbra, diffuse, specular, ambient, powerDraw, nomVoltage);
```

Comment:
(VESSEL*)(this->Vessel) is unnecessary because the ElectricLight constructor now uses a VESSEL pointer instead of OBJHANDLE. So "Vessel" alone is enough.

Reply:
should be fixed now